servers/src/main/java/tachyon/worker/block/BlockWorker.java

```diff
@@ -286,62 +196,12 @@ public void stop() throws IOException {
     mFileSystemMasterClient.close();
     // Use shutdownNow because HeartbeatThreads never finish until they are interrupted
     getExecutorService().shutdownNow();
-    mWorkerMetricsSystem.stop();
-    try {
-      mWebServer.shutdownWebServer();
-    } catch (Exception e) {
-      LOG.error("Failed to stop web server", e);
-    }
+
     mBlockDataManager.stop();
-    while (!mDataServer.isClosed() || mThriftServer.isServing()) {
-      // The reason to stop and close again is due to some issues in Thrift.
+    while (!mDataServer.isClosed()) {
```

Comment:
Do we understand why we need to close() more then once? If so, there should be a comment explaining it. If not, we should add a TODO to investigate further (and maybe a guess?).